package NYT.NScheduler.NProto;

////////////////////////////////////////////////////////////////////////////////

message TJobResources
{
    optional int64 user_slots = 1;
    optional double cpu = 2;
    optional int64 memory = 3;
    optional int64 network = 4;
    optional int32 gpu = 5;
}

message TDiskLocationQuota
{
    optional int64 disk_space = 1;
    optional int32 medium_index = 2;
}

message TDiskQuota
{
    repeated TDiskLocationQuota disk_location_quota = 1;
}

message TDiskRequest
{
    optional int64 disk_space = 1;
    optional int64 inode_count = 2;
    optional int32 medium_index = 3;
}

message TCpuRequest
{
    optional double cpu = 1;
    optional bool allow_idle_cpu_policy = 2 [default = false];
}

// TODO(babenko): consider renaming
message TJobResourcesWithQuota
{
    optional int64 user_slots = 1;
    optional double cpu = 2;
    optional int64 memory = 3;
    optional int64 network = 4;
    // COMPAT(ignat)
    optional int64 disk_quota_legacy = 5;
    optional int32 gpu = 6;
    optional TDiskQuota disk_quota = 7;
}

message TAllocationAttributes
{
    optional int64 waiting_for_resources_on_node_timeout = 1; // Duration
    optional string cuda_toolkit_version = 2;
    required bool allow_idle_cpu_policy = 3;
    required int64 port_count = 4;
    required TDiskRequest disk_request = 5;
    optional bool enable_multiple_jobs = 6;
}

////////////////////////////////////////////////////////////////////////////////
